src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc

```diff
@@ -562,6 +562,16 @@ BtreeLBAManager::update_mapping(
 {
   LOG_PREFIX(BtreeLBAManager::update_mapping);
   TRACET("laddr={}, paddr {} => {}", t, laddr, prev_addr, addr);
+#ifndef NDEBUG
+  if (addr.is_delayed() && !cache.get_checksum_needed(prev_addr)) {
```

Comment:
Hm, We need to handle two cases as below because ool writes generates delayed addr (in Segment case).

Reply:
The check is moved from `update_mappings` to `update_mapping`. Looking at the callers, `TM::rewrite_logical_extent()` was involved after the change. IIUC this is why the check becomes more complicated.

Probably we can simply drop this check, because looks the users are already validating the checksum invarients properly according to their specific scenarios, what do you think?

Reply:
The dropping is reasonable to me.